File: src/Shared/Domain/Entities/Sight.cs


namespace Domain.Entities;
public class Sight : BaseAuditableEntity
{
    public string Name { get; set; }
    public Location Location { get; set; }
    public Category Category { get; set; }
    public Guid CategoryId { get; set; }

    public Sight()
    {
        
    }

    public Sight(string name)
    {
        Name = name;
    }

    public Sight(string name, Location location) : this(name) 
    { 
        Location = location;
    }

    public ICollection<SightPhoto> SightPhotos { get; set; }
    public ICollection<SightReview> Reviews { get; set; }
    public ICollection<Tag> Tags { get; set; }

    public ICollection<Information> Information { get; set; }
}

File: src/ClientApp/src/app/core/Models/Sight.ts
import {Location} from "./Location";
import {Category} from "./category";
import {SightPhoto} from "./SightPhoto";
import {SightReview} from "./SightReview";
import {Tag} from "./Tag";
import {Guid} from "guid-typescript";
import {Information} from "./Information";

export interface Sight {
  id: Guid;
  name: string;
  description: string;
  location: Location;
  category: Category;
  sightPhotos: SightPhoto[];
  information: Information[];
  tags: Tag[];
  reviews: SightReview[];
}
